### PR TITLE
Release notes for v5.0.0

### PR DESCRIPTION
**Pull request by lcheunglci**
I left the corresponding SNI PR as TBD and will add it in as soon as the SNI PR has been created.  Please let me know if I missed anything. Thank you!

### CODE REVIEW

File: release-notes/5.0/5.0.0.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+- Updated AuthProviderInfo struct to be match the changes in native SNI for `TDS 8` server certificate validation. [#1680](https://github.com/dotnet/SqlClient/pull/1680)
+- Change default system protocol to for `TDS 8` on managed code. [#1678](https://github.com/dotnet/SqlClient/pull/1678)
+- Updated `Microsoft.Data.SqlClient.SNI` (.NET Framework dependency) and `Microsoft.Data.SqlClient.SNI.runtime` (.NET Core/Standard dependency) version to `5.0.0`. [TBD](placeholder)
+- Changed encoding UTF-7 to ASCII for SSRP Broadcast [#1671](https://github.com/dotnet/SqlClient/pull/1671)
+- Updated IdentityModel dependency from 6.8.0 to 6.21.0 and IdentityClient from 4.32.2 to 4.45.0 [#1646](https://github.com/dotnet/SqlClient/pull/1646)
+- Code health improvements [#1353](https://github.com/dotnet/SqlClient/pull/1353) [#1354](https://github.com/dotnet/SqlClient/pull/1354) [#1525](https://github.com/dotnet/SqlClient/pull/1525) [#1186](https://github.com/dotnet/SqlClient/pull/1186)
+- Update Azure Identity dependency from 1.5.0 to 1.6.0.[#1611](https://github.com/dotnet/SqlClient/pull/1611)
+- Improved Regex for SqlCommandSet [#1548] (https://github.com/dotnet/SqlClient/pull/1548)
```
Comment (ErikEJ) [inline release-notes/5.0/5.0.0.md:62]: Remove space before link

File: release-notes/5.0/5.0.0.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+- Rework on `TdsParserStateObjectManaged` with nullable annotations. [#1555] (https://github.com/dotnet/SqlClient/pull/1555)
```
Comment (ErikEJ) [inline release-notes/5.0/5.0.0.md:63]: Remove space before link

File: release-notes/5.0/5.0.0.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+- Added new attestation protocol `None` option to forgo enclave attestation when using VBS enclaves. [#1425](https://github.com/dotnet/SqlClient/pull/1425) and [#1419](https://github.com/dotnet/SqlClient/pull/1419), [Read more](#new-attestation-protocol-none)
+- Added a new AppContext switch to suppress insecure TLS warnings. [#1457](https://github.com/dotnet/SqlClient/pull/1457), [Read more](#suppress-insecure-tls-warnings)
+
+
+### Fixed
+
+- Fixed KeyNotFoundException for the FailoverPartner key on SQL servers with availability group configured. [#1614](https://github.com/dotnet/SqlClient/pull/1614)
+- Fixed small inconsistency between netcore and netfx for EncrpytionOptions [#1672](https://github.com/dotnet/SqlClient/pull/1672)
+- Fixed `Microsoft.SqlServer.Server` netcore project package reference [#1654](https://github.com/dotnet/SqlClient/pull/1654)
+- Fixed naming, order, and formatting for `SqlDiagnosticsListener` on .NET Core and .NET. [#1637] (https://github.com/dotnet/SqlClient/pull/1637)
```
Comment (ErikEJ) [inline release-notes/5.0/5.0.0.md:40]: Multiple instances of a space before the link below

File: release-notes/5.0/5.0.0.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      {  
+        Console.WriteLine("{0} = {1}", col.ColumnName, row[col]);  
+      }  
+      Console.WriteLine("============================");  
+    }  
+  }  
+```
+
+### New Attestation protocol `None`
+ new attestation protocol called `None` will be allowed in the connection string. This protocol will allow users to forgo enclave attestation for `VBS` enclaves. When this protocol is set, the enclave attestation URL property is optional.
```
Comment (David-Engel) [inline release-notes/5.0/5.0.0.md:128]:
```suggestion
A new attestation protocol called `None` is allowed in the connection string. This protocol will allow users to forgo enclave attestation for `VBS` enclaves. When this protocol is set, the enclave attestation URL property is optional.  
```

File: release-notes/5.0/5.0.0.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+Connection string example:
+
+```cs
+//Attestation protocol NONE with no URL
+"Data Source = {server}; Initial Catalog = {db}; Column Encryption Setting = Enabled; Attestation Protocol = None;"
+
+```
+
+### Suppress insecure TLS warnings
+A security warning is ouptput on the console if the TLS version less than 1.2 is used to negotiate with the server. This warning could be suppressed on SQL connection while `Encrypt = false` by enabling the following AppContext switch on the application startup:
```
Comment (David-Engel) [inline release-notes/5.0/5.0.0.md:139]:
```suggestion
A security warning is output to the console if a TLS version less than 1.2 is used to negotiate encryption with the server. This warning can be suppressed on connections where `Encrypt = false` by enabling the following AppContext switch at application startup:
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,30 @@ All notable changes to this project will be documented in this file.
...
+### Added
+
+- Added `TDS 8`  version for TDSLogin. [#1657](https://github.com/dotnet/SqlClient/pull/1657)
+- Added support for `TDS 8`. To use TDS 8, users should specify Encrypt=Strict in the connection string. [#1608](https://github.com/dotnet/SqlClient/pull/1608) [Read more](#tds-8-enhanced-security)
+
+### Fixed
+
+- Fixed null SqlBinary as rowversion. [#1688](https://github.com/dotnet/SqlClient/pull/1688)
+- Fixed KeyNotFoundException for the FailoverPartner key on SQL servers with availability group configured. [#1614](https://github.com/dotnet/SqlClient/pull/1614)
+- Fixed small inconsistency between netcore and netfx for EncrpytionOptions. [#1672](https://github.com/dotnet/SqlClient/pull/1672)
```
Comment (ErikEJ) [inline CHANGELOG.md:20]: Typo

**Comment by Kaur-Parminder** [comment]
@lcheunglci I believe latest stable update is also to be added in release-notes/README.md

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,31 @@ All notable changes to this project will be documented in this file.
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Stable release 5.0.0] - 2022-08-05
+
+This update brings the below changes over the previous release:
+
+### Added
+
+- Added `TDS 8` version for TDSLogin. [#1657](https://github.com/dotnet/SqlClient/pull/1657)
+- Added support for `TDS 8`. To use TDS 8, users should specify Encrypt=Strict in the connection string. [#1608](https://github.com/dotnet/SqlClient/pull/1608) [Read more](#tds-8-enhanced-security)
```
Comment (cheenamalhotra) [inline CHANGELOG.md:14]: Casually reviewing.. 👀 You may remove the Read more section from here 🙂

**Review by David-Engel — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

File: release-notes/5.0/5.0.0.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+- Dropped support for .NET Framework 4.6.1 [#1574](https://github.com/dotnet/SqlClient/pull/1574)
+
+
+### Added
+
+- Added `TDS 8`  version for TDSLogin. [#1657](https://github.com/dotnet/SqlClient/pull/1657)
+- Added support for `TDS 8`. To use TDS 8, users should specify Encrypt=Strict in the connection string. [#1608](https://github.com/dotnet/SqlClient/pull/1608) [Read more](#tds-8-enhanced-security)
+- Added support for specifying Server SPN and Failover Server SPN on the connection. [#1607](https://github.com/dotnet/SqlClient/pull/1607) [Read more](#server-spn)
+- Added support for aliases when targeting .NET Core on Windows. [#1588](https://github.com/dotnet/SqlClient/pull/1588) [Read more](#support-for-aliases)
+- Added SqlDataSourceEnumerator. [#1430](https://github.com/dotnet/SqlClient/pull/1430), [Read more](#sql-data-source-enumerator-support)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:30]:
```suggestion
- Added support for `SqlDataSourceEnumerator` on Windows. [#1430](https://github.com/dotnet/SqlClient/pull/1430), [Read more](#sql-data-source-enumerator-support)
```

File: release-notes/5.0/5.0.0.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+- Added new attestation protocol `None` option to forgo enclave attestation when using VBS enclaves. [#1425](https://github.com/dotnet/SqlClient/pull/1425) and [#1419](https://github.com/dotnet/SqlClient/pull/1419), [Read more](#new-attestation-protocol-none)
+- Added a new AppContext switch to suppress insecure TLS warnings. [#1457](https://github.com/dotnet/SqlClient/pull/1457), [Read more](#suppress-insecure-tls-warnings)
+
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:33]: Extra line.
```suggestion
```

File: release-notes/5.0/5.0.0.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+### Fixed
+
+- Fixed null SqlBinary as rowversion. [#1688](https://github.com/dotnet/SqlClient/pull/1688)
+- Fixed KeyNotFoundException for the FailoverPartner key on SQL servers with availability group configured. [#1614](https://github.com/dotnet/SqlClient/pull/1614)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:38]:
```suggestion
- Fixed **KeyNotFoundException** for the `FailoverPartner` key on SQL servers with availability group configured. [#1614](https://github.com/dotnet/SqlClient/pull/1614)
```

File: release-notes/5.0/5.0.0.md [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+- Fixed naming, order, and formatting for `SqlDiagnosticsListener` on .NET Core and .NET. [#1637](https://github.com/dotnet/SqlClient/pull/1637)
+- Fixed NullReferenceException during Azure Active Directory authentication. [#1625](https://github.com/dotnet/SqlClient/pull/1625)
+- Added CommandText length validation when using stored procedure command types. [#1484](https://github.com/dotnet/SqlClient/pull/1484)
+- Fixed `GetSchema("StructuredTypeMembers")` to return correct schema information. [#1500](https://github.com/dotnet/SqlClient/pull/1500), [#1639](https://github.com/dotnet/SqlClient/pull/1639)
+- Fixed NullReferenceException when using `SqlDependency.Start` against an Azure SQL Database. [#1294](https://github.com/dotnet/SqlClient/pull/1294)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:43]:
```suggestion
- Fixed **NullReferenceException** when using `SqlDependency.Start` against an Azure SQL Database. [#1294](https://github.com/dotnet/SqlClient/pull/1294)
```

File: release-notes/5.0/5.0.0.md [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+- Send the correct retained transaction descriptor in the MARS TDS Header when there is no current transaction on .NET 5+ and .NET Core. [#1624](https://github.com/dotnet/SqlClient/pull/1624)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:44]:
```suggestion
- Fixed transaction descriptor in the MARS TDS Header when there is no current transaction on .NET 5+ and .NET Core. [#1624](https://github.com/dotnet/SqlClient/pull/1624)
```

File: release-notes/5.0/5.0.0.md [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+- Parallelize SSRP requests on Linux and macOS when MultiSubNetFailover is specified. [#1578](https://github.com/dotnet/SqlClient/pull/1578)
+- Fixed connection failure by skipping Certificate Revocation List (CRL) check during authentication. [#1559](https://github.com/dotnet/SqlClient/pull/1559)
+- Fixed thread safety issue for `GetEnclaveProvider` by converting dictionary to concurrent dictionary. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
+
+### Changed
+
+- Updated AuthProviderInfo struct to be match the changes in native SNI for `TDS 8` server certificate validation. [#1680](https://github.com/dotnet/SqlClient/pull/1680)
+- Change default system protocol to for `TDS 8` on managed code. [#1678](https://github.com/dotnet/SqlClient/pull/1678)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:52]:
```suggestion
- Updated default system protocol for `TDS 8` on managed code. [#1678](https://github.com/dotnet/SqlClient/pull/1678)
```

File: release-notes/5.0/5.0.0.md [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+- Fixed `GetSchema("StructuredTypeMembers")` to return correct schema information. [#1500](https://github.com/dotnet/SqlClient/pull/1500), [#1639](https://github.com/dotnet/SqlClient/pull/1639)
+- Fixed NullReferenceException when using `SqlDependency.Start` against an Azure SQL Database. [#1294](https://github.com/dotnet/SqlClient/pull/1294)
+- Send the correct retained transaction descriptor in the MARS TDS Header when there is no current transaction on .NET 5+ and .NET Core. [#1624](https://github.com/dotnet/SqlClient/pull/1624)
+- Parallelize SSRP requests on Linux and macOS when MultiSubNetFailover is specified. [#1578](https://github.com/dotnet/SqlClient/pull/1578)
+- Fixed connection failure by skipping Certificate Revocation List (CRL) check during authentication. [#1559](https://github.com/dotnet/SqlClient/pull/1559)
+- Fixed thread safety issue for `GetEnclaveProvider` by converting dictionary to concurrent dictionary. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
+
+### Changed
+
+- Updated AuthProviderInfo struct to be match the changes in native SNI for `TDS 8` server certificate validation. [#1680](https://github.com/dotnet/SqlClient/pull/1680)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:51]:
```suggestion
- Updated `AuthProviderInfo` struct to be matched the changes in native SNI for `TDS 8` server certificate validation. [#1680](https://github.com/dotnet/SqlClient/pull/1680)
```

File: release-notes/5.0/5.0.0.md [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+- Change default system protocol to for `TDS 8` on managed code. [#1678](https://github.com/dotnet/SqlClient/pull/1678)
+- Updated `Microsoft.Data.SqlClient.SNI` (.NET Framework dependency) and `Microsoft.Data.SqlClient.SNI.runtime` (.NET Core/Standard dependency) version to `5.0.0`. [#1608](https://github.com/dotnet/SqlClient/pull/1608)
+- Changed encoding UTF-7 to ASCII for SSRP Broadcast. [#1671](https://github.com/dotnet/SqlClient/pull/1671)
+- Updated IdentityModel dependency from 6.8.0 to 6.21.0 and IdentityClient from 4.32.2 to 4.45.0. [#1646](https://github.com/dotnet/SqlClient/pull/1646)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:55]:
```suggestion
- Updated `IdentityModel` dependency from 6.8.0 to 6.21.0 and `IdentityClient` from 4.32.2 to 4.45.0. [#1646](https://github.com/dotnet/SqlClient/pull/1646)
```

File: release-notes/5.0/5.0.0.md [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+- Update Azure Identity dependency from 1.5.0 to 1.6.0. [#1611](https://github.com/dotnet/SqlClient/pull/1611)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:56]:
```suggestion
- Updated **Azure Identity** dependency from 1.5.0 to 1.6.0. [#1611](https://github.com/dotnet/SqlClient/pull/1611)
```

File: release-notes/5.0/5.0.0.md [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+- Improved Regex for SqlCommandSet. [#1548](https://github.com/dotnet/SqlClient/pull/1548)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:57]:
```suggestion
- Improved Regex for `SqlCommandSet`. [#1548](https://github.com/dotnet/SqlClient/pull/1548)
```

File: release-notes/5.0/5.0.0.md [hunk 18]
```diff
@@ -0,0 +1,204 @@
...
+- Rework `TdsParserStateObjectManaged` with nullable annotations. [#1555](https://github.com/dotnet/SqlClient/pull/1555)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:58]: This can be moved to the various code improvements.

File: release-notes/5.0/5.0.0.md [hunk 19]
```diff
@@ -0,0 +1,204 @@
...
+- Adjust the default ConnectRetryCount against Azure Synapse OnDemand endpoints. [#1626](https://github.com/dotnet/SqlClient/pull/1626)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:59]:
```suggestion
- Adjust the default **ConnectRetryCount** against Azure Synapse OnDemand endpoints. [#1626](https://github.com/dotnet/SqlClient/pull/1626)
```

File: release-notes/5.0/5.0.0.md [hunk 20]
```diff
@@ -0,0 +1,204 @@
...
+- Updated `Azure.Identity`  version to `1.5.0` and `Microsoft.Identity.Client` version to `4.30.1`. [#1462](https://github.com/dotnet/SqlClient/pull/1462)
+- Replaced AlwaysEncryptedAttestationException with SqlException. [#1515](https://github.com/dotnet/SqlClient/pull/1515)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:61]:
```suggestion
- Replaced `AlwaysEncryptedAttestationException` with `SqlException`. [#1515](https://github.com/dotnet/SqlClient/pull/1515)
```

File: release-notes/5.0/5.0.0.md [hunk 21]
```diff
@@ -0,0 +1,204 @@
...
+- Improved error message when adding wrong type to SqlParameterCollection. [#1547](https://github.com/dotnet/SqlClient/pull/1547)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:62]:
```suggestion
- Improved error message when adding wrong type to `SqlParameterCollection`. [#1547](https://github.com/dotnet/SqlClient/pull/1547)
```

File: release-notes/5.0/5.0.0.md [hunk 22]
```diff
@@ -0,0 +1,204 @@
...
+- Modernized style in ValueUtilSmi. [#1351](https://github.com/dotnet/SqlClient/pull/1351)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:63]: If it's only a style change, it should be moved to the various code improvements section.

File: release-notes/5.0/5.0.0.md [hunk 23]
```diff
@@ -0,0 +1,204 @@
...
+- Changed SQL server codenames to version names in the code. [#1439](https://github.com/dotnet/SqlClient/pull/1439)
+- Prevented subtype generation in project files. [#1452](https://github.com/dotnet/SqlClient/pull/1452)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:65]: It can be moved to the various code improvements section.

File: release-notes/5.0/5.0.0.md [hunk 24]
```diff
@@ -0,0 +1,204 @@
...
+- Changed `Array.Copy` to `Buffer.BlockCopy` for byte arrays. [#1366](https://github.com/dotnet/SqlClient/pull/1366)
+- Changed files in csproj to be alphabetically sorted in netfx and netcore. [#1364](https://github.com/dotnet/SqlClient/pull/1364)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:67]: It can be moved to the various code improvements section.

File: release-notes/5.0/5.0.0.md [hunk 25]
```diff
@@ -0,0 +1,204 @@
...
+- Sqlstream, SqlInternalTransaction and MetaDataUtilsSmi are moved to shared folder. [#1337](https://github.com/dotnet/SqlClient/pull/1337), [#1346](https://github.com/dotnet/SqlClient/pull/1346) and [#1339](https://github.com/dotnet/SqlClient/pull/1339)
```
Comment (DavoudEshtehari) [inline release-notes/5.0/5.0.0.md:68]: These can be moved to the various code improvements section.

File: release-notes/5.0/5.0.0.md [hunk 26]
```diff
@@ -0,0 +1,199 @@
...
+  - (enum) Microsoft.Data.SqlClient.Server.DataAccessKind -> Microsoft.SqlServer.Server.DataAccessKind
+  - (enum) Microsoft.Data.SqlClient.Server.Format -> Microsoft.SqlServer.Server.Format
+  - (enum) Microsoft.Data.SqlClient.Server.SystemDataAccessKind -> Microsoft.SqlServer.Server.SystemDataAccessKind
+- Dropped support for .NET Framework 4.6.1 [#1574](https://github.com/dotnet/SqlClient/pull/1574)
+
+
+### Added
+
+- Added `TDS 8`  version for TDSLogin. [#1657](https://github.com/dotnet/SqlClient/pull/1657)
+- Added support for `TDS 8`. To use TDS 8, users should specify Encrypt=Strict in the connection string. [#1608](https://github.com/dotnet/SqlClient/pull/1608) [Read more](#tds-8-enhanced-security)
```
Comment (JRahnama) [inline release-notes/5.0/5.0.0.md:27]: Just a minor re-order
```suggestion
- Added support for `TDS 8`. To use TDS 8, users should specify Encrypt=Strict in the connection string. [#1608](https://github.com/dotnet/SqlClient/pull/1608) [Read more](#tds-8-enhanced-security)
- Added `TDS 8`  version for TDSLogin. [#1657](https://github.com/dotnet/SqlClient/pull/1657)
```

File: release-notes/5.0/5.0.0.md [hunk 27]
```diff
@@ -0,0 +1,199 @@
...
+
+
+### TDS 8 Enhanced Security
+
+To use TDS 8, specify Encrypt=Strict in the connection string. Strict mode disables TrustServerCertificate (always treated as False in Strict mode). HostNameInCertificate has been added to help some Strict mode scenarios. TDS 8 begins and continues all server communication inside a secure, encrypted TLS connection.
+
+New Encrypt values have been added to clarify connection encryption behavior. Encrypt=Mandatory is equavalent to Encrypt=True and encrypts connections during the TDS connection negotiation. Encrypt=Optional is equivalent to Encrypt=False and only encrypts the connection if the server tells the client that encryption is required during the TDS connection negotiation.
```
Comment (JRahnama) [inline release-notes/5.0/5.0.0.md:71]: typo equivalent
```suggestion
New Encrypt values have been added to clarify connection encryption behavior. Encrypt=Mandatory is equivalent to Encrypt=True and encrypts connections during the TDS connection negotiation. Encrypt=Optional is equivalent to Encrypt=False and only encrypts the connection if the server tells the client that encryption is required during the TDS connection negotiation.
```

File: release-notes/5.0/5.0.0.md [hunk 28]
```diff
@@ -0,0 +1,199 @@
...
+- .NET Framework 4.6.2+ (Windows x86, Windows x64)
+- .NET Core 3.1+ (Windows x86, Windows x64, Windows ARM64, Windows ARM, Linux, macOS)
+- .NET Standard 2.0+ (Windows x86, Windows x64, Windows ARM64, Windows ARM, Linux, macOS)
+
+### Dependencies
+
+#### .NET Framework
+
+- Microsoft.Data.SqlClient.SNI.runtime 5.0.0
+- Azure.Identity 1.6.0.0
```
Comment (JRahnama) [inline release-notes/5.0/5.0.0.md:144]: where are the extra zeros coming from?
Azure.Identity for example the version is 1.6.0 and for the rest below it.

Comment (lcheunglci) [reply release-notes/5.0/5.0.0.md]: revised with the same entries shown in the nuget package explorer

**Review by JRahnama — APPROVED** [review]

**Pull request merged**